tests/ConversationTranslatorTests.ts

```diff
@@ -84,6 +86,401 @@ describe("conversation constructor tests", () => {
+
+        const c: sdk.Conversation = sdk.Conversation.createConversationAsync(config, (() => {
+            const ct: sdk.ConversationTranslator = new sdk.ConversationTranslator();
+            if (endpointHost !== "") { ct.properties.setProperty(sdk.PropertyId.ConversationTranslator_Host, endpointHost); }
+            if (speechEndpointHost !== "") { ct.properties.setProperty(sdk.PropertyId[sdk.PropertyId.SpeechServiceConnection_Host], speechEndpointHost); }
+
+            ct.canceled = ((s: sdk.ConversationTranslator, e: sdk.ConversationTranslationCanceledEventArgs) => {
+                done();
+            });
+            ct.participantsChanged = ((s: sdk.ConversationTranslator, e: sdk.ConversationParticipantsChangedEventArgs) => {
+                if (e.reason === sdk.ParticipantChangedReason.JoinedConversation) {
+                    participantsCount++;
+                    if (participantsCount === 1) {
+                        joinParticipant(c.conversationId);
+                    } else {
+                        c.muteAllParticipantsAsync();
+                    }
+                } else if (e.reason === sdk.ParticipantChangedReason.Updated) {
+                    commandCount++;
+                    if (e.participants.length > 0) {
+                        // only the participants should be updated by a mute all command
+                        expect(e.participants[0].isMuted).toEqual(isMuted);
+                        isMuted = !isMuted;
+                        c.unmuteAllParticipantsAsync();
+                        done();
+                    }
+                }
+            });
+
+            objsToClose.push(ct);
+
+            c.startConversationAsync(() => {
+                ct.joinConversationAsync(c, "Host",
+                    (() => {
+                        // continue
+                    }),
+                    ((error: any) => {
+                        done.fail();
+                    }));
+            });
+        }),
+        ((error: any) => {
+            done.fail();
+        }));
+        objsToClose.push(c);
+
+        function joinParticipant(code: string): void {
+            // join as a participant
+            const ctP: sdk.ConversationTranslator = new sdk.ConversationTranslator();
+            if (endpointHost !== "") { ctP.properties.setProperty(sdk.PropertyId.ConversationTranslator_Host, endpointHost); }
+            if (speechEndpointHost !== "") { ctP.properties.setProperty(sdk.PropertyId[sdk.PropertyId.SpeechServiceConnection_Host], speechEndpointHost); }
+            objsToClose.push(ctP);
+            ctP.joinConversationAsync(code, "mute me", "en-US",
+                    (() => {
+                        // continue
+                    }),
+                    ((error: any) => {
+                        done.fail();
+                    }));
+        }
+
+        WaitForCondition(() => (commandCount > 2), done);
```

Comment:
>done [](start = 51, length = 4)

done() is being called in too many places. There can be many calls to done.fail, but there should only be one way done() is called to end the test, when everything is done.

Reply:
I've removed some of the extra dones.